lib/temporal/activity.rb

```diff
@@ -4,10 +4,18 @@ def self.activity_name(new_name)
       @activity_name = new_name
     end
 
+    def self.shielded
```

Comment:
Can we follow Ruby idioms here and mark this as `shielded!`. I also like python's SDK name `shield_thread_cancel_exception` (maybe without thread) where it is clear that the cancellation error will be skipped.

Reply:
Oh, didn't know they use `!` in this case, do you have examples? I'm fine with both `shielded!` and `shield_cancel_exception!`.

@cretz, @bergundy do you guys have a preference?

Reply:
> python's SDK name

That is to disambiguate against [asyncio.shield](https://docs.python.org/3/library/asyncio-task.html#asyncio.shield) which most would use if they are using async activities. In Python we have a difference between async and sync threaded and sync multiprocess activities, and that method name is to clarify it's only for sync threaded.

Reply:
@antstorm - if this will always be the way code can shield from cancellation exceptions, I vote `shielded!`. If there is a difference between how this shielded will work vs, say, one using fibers or something then it'd be worth qualifying. Basically I only care from the callers' POV. If they don't have to care, then don't qualify.

Reply:
from a caller's perspective at least from my side `shielded` is a bit ambiguous. Shielded from what? This is the reason I prefer the `cancellation` to be present in the naming.

Last but not least the cancellation also comes from the server since the `workflows` requests a cancellation right? So I think this should be reflected in the naming as per: https://docs.temporal.io/concepts/what-is-an-activity-execution#request-cancellation

Reply:
In the Python standard library this is called `shield` and in Temporal Go workflows this is `NewDisconnectedContext`. All other cases are using "cancellation scopes" (which does have the word cancellation) or try/catch knowing it throws from heartbeat. I don't think Temporal's use of "shielded" would be ambiguous with any use of that term in Temporal.

Reply:
I vote not. Don't have to Go, don't have to in Python (for normal async activities), and "shield" is not ambiguous with any other term.

Reply:
I side with @cretz here, since there's no other type of shielding we're just making this overly explicit by adding `_from_cancellation`. Once you're checked the docs or source once you'd know exactly what that means.

A different angle would be to use something like `disallow_cancellations` or `prevent_cancellations`, but I think `shielded` is nice and concise.

Reply:
Renamed — https://github.com/temporalio/sdk-ruby/pull/90/commits/4f42ae8d359395988cb91fefd81509730fb2a3e2